packages/next/src/withTRPC.tsx

```diff
@@ -32,7 +32,7 @@ import {
 } from 'next/dist/shared/lib/utils';
 import { NextRouter } from 'next/router';
 import React, { createElement, useState } from 'react';
-import ssrPrepass from 'react-ssr-prepass';
+import { renderToPipeableStream } from 'react-dom/server';
```

Comment:
Not super familiar with what stuff got introduced in react 18 but isn't this one of them? in that case we'd now have peerDep of react^18 instead of react^16?